### ISSUE TITLE
websocket_disconnect not called

### ISSUE THREAD

**Issue by brianmay** [django/channels#892]
When a WebSocket is closed, I see:

```
[2018/02/09 14:22:23] WebSocket DISCONNECT /ws/ [127.0.0.1:38682]
```

But I don't see any call to the ``websocket_disconnect`` method of the consumer. In fact, I don't see any call to the ``dispatch`` method.

My suspicion is the problem lies in Daphne. Where it calls ``self.server.discard_protocol``, which I belive (if I understand this correctly) kills the consumer, and then adds a message to the consumer's queue. Which of course will never get read.

**Comment by andrewgodwin** [comment]
This was issue https://github.com/django/daphne/issues/152, fixed in release 2.0.3. Did you upgrade?

**Comment by Avinash-Raj** [comment]
I experience the same issue in `daphne==2.2.3`
